sycl/include/sycl/ext/oneapi/experimental/annotated_ptr/annotated_ptr.hpp

```diff
@@ -118,6 +118,45 @@ class annotated_ref<T, detail::properties_t<Props...>> {
   PROPAGATE_OP(|)
   PROPAGATE_OP(<<)
   PROPAGATE_OP(>>)
+#undef PROPAGATE_OP
+
+  template <class O> struct get_T {
```

Comment:
helper types which aren't part of the public API should be in detail namespace.

nit: let's give this a better name e.g. `remove_ann_ref_impl`